### ISSUE TITLE
`Layout/LineEndStringConcatenationIndentation` - undefined method `parent' for nil:NilClass

### ISSUE THREAD

**Issue by snutij** [rubocop/rubocop#11994]
Hello 👋 

Given a file `foo.rb`: 
```ruby
x = %



```

Which is a valid syntax: 
```
ruby -c foo.rb
Syntax OK
```

--------

## Expected behavior

Should not trigger error.

## Actual behavior

`rubocop -d --only Layout/LineEndStringConcatenationIndentation foo.rb`
 
```
An error occurred while Layout/LineEndStringConcatenationIndentation cop was inspecting foo.rb:1:4.
undefined method `parent' for nil:NilClass
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/layout/line_end_string_concatenation_indentation.rb:127:in `base_column'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/layout/line_end_string_concatenation_indentation.rb:121:in `check_indented'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/layout/line_end_string_concatenation_indentation.rb:90:in `on_dstr'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/commissioner.rb:107:in `public_send'
```

## Steps to reproduce the problem
In `LineEndStringConcatenationIndentation` the `on_dstr` receive a node: 

```
(rdbg) node
s(:dstr)
```

Then node.children result in: 

```
(ruby) node.children
[]
```

So the `base_column` method receive a `[][0]` which is `nil`.

https://github.com/rubocop/rubocop/blob/aaf1c3f6285e5e9e84d43c37944a7faaacb3bf60/lib/rubocop/cop/layout/line_end_string_concatenation_indentation.rb#L83-L93

https://github.com/rubocop/rubocop/blob/aaf1c3f6285e5e9e84d43c37944a7faaacb3bf60/lib/rubocop/cop/layout/line_end_string_concatenation_indentation.rb#L120-L124

https://github.com/rubocop/rubocop/blob/aaf1c3f6285e5e9e84d43c37944a7faaacb3bf60/lib/rubocop/cop/layout/line_end_string_concatenation_indentation.rb#L126-L133

## RuboCop version

```
bundle exec rubocop -V
1.53.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-minitest 0.31.0
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.7.0
```
